packages/expo-cli/src/commands/upgrade.ts

```diff
@@ -0,0 +1,217 @@
+): Promise<DependencyList> {
+  let result: DependencyList = {};
+
+  // Get the supported react-native version.
+  // * We can't update the react-native version if the version data isn't published
+  if (targetSdkVersion) {
+    result['react-native'] = `https://github.com/expo/react-native/archive/${
+      targetSdkVersion.expoReactNativeTag
+    }.tar.gz`;
+
+    // React version apparently is optional in SDK version data
+    if (targetSdkVersion.facebookReactVersion) {
+      result['react'] = targetSdkVersion.facebookReactVersion;
+    }
+
+    // - TODO: how do we know what version of react-native-web, react-dom to use?
+    // - TODO: we need to track supported devDependencies versions somewhere programmatic too
+  } else {
+    console.log(`Supported React Native version unknown, please update it manually.`);
+  }
+
+  // Get the updated version for any bundled modules
+  let { exp, pkg } = await ConfigUtils.readConfigJsonAsync(projectRoot);
+  let bundledNativeModules = (await JsonFile.readAsync(
+    ConfigUtils.resolveModule('expo/bundledNativeModules.json', projectRoot, exp)
+  )) as DependencyList;
+
+  let { dependencies } = pkg;
+  Object.keys(bundledNativeModules).forEach(name => {
+    if (dependencies[name]) {
+      result[name] = bundledNativeModules[name];
+    }
+  });
+
+  return result;
+}
+
+async function upgradeAsync(requestedSdkVersion: string | null, options: Options) {
+  let { projectRoot, workflow } = await findProjectRootAsync(process.cwd());
+  let { exp } = await ConfigUtils.readConfigJsonAsync(projectRoot);
+
+  if (Versions.lteSdkVersion(exp, '32.0.0')) {
+    log.error('The upgrade command is only available on SDK 33 and higher. Please refer to https://docs.expo.io/versions/latest/workflow/upgrading-expo-sdk-walkthrough/ for upgrade instructions.');
+    return;
+  }
+
+  // Can't upgrade if we don't have a SDK version
+  if (!exp.sdkVersion) {
+    log.error('No sdkVersion field is present in app.json, cannot upgrade project.');
+    return;
+  }
+
+  let currentSdkVersionString = exp.sdkVersion;
+  let sdkVersions = await Versions.sdkVersionsAsync();
+  let latestSdkVersion = await Versions.newestSdkVersionAsync();
+  let latestSdkVersionString = latestSdkVersion.version;
+  let targetSdkVersionString =
+    maybeFormatSdkVersion(requestedSdkVersion) || latestSdkVersion.version;
+  let targetSdkVersion = sdkVersions[targetSdkVersionString];
+
+  // TODO: figure out how this should work for bare workflow
+  if (workflow === 'bare') {
+    console.log('This command is currently only supported on the managed workflow.');
+    return;
+  }
+
+  // Bail out early if people are trying to update to the current version
+  if (targetSdkVersionString === currentSdkVersionString) {
+    console.log(
+      'You are already using the latest SDK version. Follow https://blog.expo.io for new release information.'
+    );
+    return;
```

Comment:
We should throw an error (e.g. `CommandError`) to make the return value of the command non-zero.

Reply:
i think in this case i'd expect it to exit with status 0 because, while it didn't upgrade, the command ran successfully. we already were on latest version so no need to change anything